### ISSUE TITLE
SHELL instruction isn't applied at image build time

### ISSUE THREAD

**Issue by EricHripko** [containers/buildah#1581]
**Description**

**Steps to reproduce the issue:**
1. Use the following `Dockerfile`
```
FROM docker.io/ubuntu
SHELL ["/usr/bin/this/does/not/exist"]
RUN echo "Hello world"
```
2. Build an image `buildah bud --format=docker .`

**Describe the results you received:**
Image build succeeds. We can, therefore, conclude that shell wasn't actually changed.
```
STEP 1: FROM docker.io/ubuntu
STEP 2: SHELL ["/usr/bin/this/does/not/exist"]
STEP 3: RUN echo "Hello world"
Hello world
STEP 4: COMMIT
Getting image source signatures
Copying blob 762d8e1a6054 skipped: already exists
Copying blob e45cfbc98a50 skipped: already exists
Copying blob d60e01b37e74 skipped: already exists
Copying blob b57c79f4a9f3 skipped: already exists
Copying blob 5f70bf18a086 skipped: already exists
Copying config 9e1fa82b3f done
Writing manifest to image destination
Storing signatures
--> 9e1fa82b3f37ed3b1d6b0a283133550f65324cbbf93c04ba6257d4d53b7e2085
```

**Describe the results you expected:**
Expected `buildah` to attempt to use the `SHELL` defined and, as a result, fail to build the image.
Docker output to compare:
```
$ docker build -t .
Step 1/3 : FROM docker.io/ubuntu
 ---> 94e814e2efa8
Step 2/3 : SHELL ["/usr/bin/this/does/not/exist"]
 ---> Running in dc5c819df407
Removing intermediate container dc5c819df407
 ---> 0735bcf1de70
Step 3/3 : RUN echo "Hello world"
 ---> Running in a3f4d8c42eb3
OCI runtime create failed: container_linux.go:348: starting container process caused "exec: \"/usr/bin/this/does/not/exist\": stat /usr/bin/this/does/not/exist: no such file or directory": unknown
```

**Output of `rpm -q buildah` or `apt list buildah`:**

```
Listing... Done
buildah/xenial,now 1.8-1~dev~ubuntu16.04.2~ppa5 amd64 [installed]
```

**Output of `buildah version`:**

```
Version:         1.8-dev
Go Version:      go1.10.4
Image Spec:      1.0.0
Runtime Spec:    1.0.0
CNI Spec:        0.4.0
libcni Version:
Git Commit:
Built:           Sat Apr 13 16:42:38 2019
OS/Arch:         linux/amd64
```

**Output of `podman version` if reporting a `podman build` issue:**

```
N/A
```

**Output of `cat /etc/*release`:**

```
DISTRIB_ID=Ubuntu
DISTRIB_RELEASE=16.04
DISTRIB_CODENAME=xenial
DISTRIB_DESCRIPTION="Ubuntu 16.04.6 LTS"
NAME="Ubuntu"
VERSION="16.04.6 LTS (Xenial Xerus)"
ID=ubuntu
ID_LIKE=debian
PRETTY_NAME="Ubuntu 16.04.6 LTS"
VERSION_ID="16.04"
HOME_URL="http://www.ubuntu.com/"
SUPPORT_URL="http://help.ubuntu.com/"
BUG_REPORT_URL="http://bugs.launchpad.net/ubuntu/"
VERSION_CODENAME=xenial
UBUNTU_CODENAME=xenial
```

**Output of `uname -a`:**

```
Linux scw-35eb3f 4.5.7-docker-4 #1 SMP Tue Jul 12 09:41:14 UTC 2016 x86_64 x86_64 x86_64 GNU/Linux
```

**Output of `cat /etc/containers/storage.conf`:**

```
# storage.conf is the configuration file for all tools
# that share the containers/storage libraries
# See man 5 containers-storage.conf for more information

# The "container storage" table contains all of the server options.
[storage]

# Default Storage Driver
driver = "overlay"

# Temporary storage location
runroot = "/var/run/containers/storage"

# Primary read-write location of container storage
graphroot = "/var/lib/containers/storage"

[storage.options]
# AdditionalImageStores is used to pass paths to additional read-only image stores
# Must be comma separated list.
additionalimagestores = [
]

# Size is used to set a maximum size of the container image.  Only supported by
# certain container storage drivers (currently overlay, zfs, vfs, btrfs)
size = ""

# OverrideKernelCheck tells the driver to ignore kernel checks based on kernel version
override_kernel_check = "true"
```

**Comment by EricHripko** [comment]
```
DEBU[0000] Parsed Step: {Env:[PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin] Command:shell Args:[/usr/bin/this/does/not/exist] Flags:[] Attrs:map[json:true] Message:SHELL /usr/bin/this/does/not/exist Original:SHELL ["/usr/bin/this/does/not/exist"]}
STEP 2: SHELL ["/usr/bin/this/does/not/exist"]
DEBU[0000] Parsed Step: {Env:[PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin] Command:run Args:[echo "Hello world"] Flags:[] Attrs:map[] Message:RUN echo "Hello world" Original:RUN echo "Hello world"}
STEP 3: RUN echo "Hello world"
DEBU[0000] RUN imagebuilder.Run{Shell:true, Args:[]string{"echo \"Hello world\""}}, docker.Config{Hostname:"a6bc593789a7", Domainname:"", User:"", Memory:0, MemorySwap:0, MemoryReservation:0, KernelMemory:0, CPUShares:0, CPUSet:"", PortSpecs:[]string(nil), ExposedPorts:map[docker.Port]struct {}{}, PublishService:"", StopSignal:"", StopTimeout:0, Env:[]string{"PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin"}, Cmd:[]string{"/bin/bash"}, Shell:[]string{"/usr/bin/this/does/not/exist"}, Healthcheck:(*docker.HealthConfig)(nil), DNS:[]string(nil), Image:"", Volumes:map[string]struct {}{}, VolumeDriver:"", WorkingDir:"", MacAddress:"", Entrypoint:[]string{}, SecurityOpts:[]string(nil), OnBuild:[]string{}, Mounts:[]docker.Mount(nil), Labels:map[string]string{}, AttachStdin:false, AttachStdout:false, AttachStderr:false, ArgsEscaped:false, Tty:false, OpenStdin:false, StdinOnce:false, NetworkDisabled:false, VolumesFrom:""}
```
Seems like `buildah` is aware that there's a shell change, but isn't utilising it.

**Comment by EricHripko** [comment]
I've tracked down the issue to the `if` condition in [StageExecutor.Run](https://github.com/containers/buildah/blob/master/imagebuildah/build.go#L591). I've managed to resolve the issue by modifying the code in the following way:
```go
	if run.Shell {
		if len(config.Shell) > 0 && s.builder.Format == buildah.Dockerv2ImageManifest {
			args = append(config.Shell, args...)
		} else {
			args = append([]string{"/bin/sh", "-c"}, args...)
		}
	}
```
This functionality is only enabled when `--format=docker` is specified. In my tests this seems to have worked both when `SHELL` was specified and omitted (with Docker format both enabled and disabled). Should I create PR for this?
